suricata/update/main.py

```diff
@@ -1063,12 +1063,10 @@ def _main():
                 rulemap[rule.id] = fltr.run(rule)
                 drop_count += 1
 
-    # Apply modify filters.
-    for fltr in modify_filters:
-        for key, rule in rulemap.items():
+        for fltr in modify_filters:
             if fltr.match(rule):
                 new_rule = fltr.run(rule)
-                if new_rule and new_rule.format() != rule.format():
```

Comment:
I may be missing something here but it seems like the modify filter can return two values:
1. Original rule in case the filter resulted in an invalid rule after modification.
2. Modified rule in case everything happened successfully.

This check made sure that the modified rule was valid and only a valid rule gets added to the rulemap.
@jasonish is this OK?

Reply:
This was a way to detect failed modify filters adding to the count, and I think its quite an expensive check to be doing now that I'm seeing more examples of people modifying 1000s, or even 10000+ rules. A log error is still printed. So I think its OK.